dartagnan/src/main/java/com/dat3m/dartagnan/program/event/CondJump.java

```diff
@@ -168,9 +164,7 @@ public CondJump getCopy(){
 
     @Override
     protected RecursiveFunction<Integer> compileRecursive(Arch target, int nextId, Event predecessor, int depth) {
-        if(successor == null){
-            throw new MalformedProgramException("Malformed CondJump event");
-        }
+    	Preconditions.checkNotNull(successor, "Malformed CondJump event");
```

Comment:
Either use `MalformedProgramException` or use `Preconditions.checkState`.
`Preconditions.checkNotNull` and `Preconditions.checkArgument` are both intended to validate input parameters only.